devel/githubContrib_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Simple tests for non github part of githubContrib.go
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"log"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+func CheckEqual(t *testing.T, msg interface{}, expected interface{}, actual interface{}) {
+	if expected != actual {
+		_, file, line, _ := runtime.Caller(1)
+		file = file[strings.LastIndex(file, "/")+1:]
+		fmt.Printf("%s:%d mismatch!\nexpected:\n%+v\nactual:\n%+v\nfor %+v\n", file, line, expected, actual, msg)
```

Comment:
t.FatalF aborts the test (wouldn't collect further errors) and would have the wrong line number (line inside `CheckEqual` instead of the line of the caller)